File: Tyche.Android/data/bet/src/main/java/com/felipearpa/tyche/data/bet/application/GetPendingPoolGamblerBetsUseCase.kt
package com.felipearpa.tyche.data.bet.application

import com.felipearpa.tyche.data.bet.domain.PoolGamblerBetRepository
import javax.inject.Inject

class GetPendingPoolGamblerBetsUseCase @Inject constructor(private val poolGamblerBetRepository: PoolGamblerBetRepository) {
    suspend fun execute(
        poolId: String,
        gamblerId: String,
        next: String? = null,
        searchText: String? = null
    ) =
        poolGamblerBetRepository.getPendingPoolGamblerBets(
            poolId = poolId,
            gamblerId = gamblerId,
            next = next,
            searchText = searchText
        )
}

File: iOS/DataBet/Sources/DataBet/Application/GetPendingPoolGamblerBetsUseCase.swift
import Core

public class GetPendingPoolGamblerBetsUseCase {
    private let poolGamblerBetRepository: PoolGamblerBetRepository
    
    public init(poolGamblerBetRepository: PoolGamblerBetRepository) {
        self.poolGamblerBetRepository = poolGamblerBetRepository
    }
    
    public func execute(
        poolId: String,
        gamblerId: String,
        next: String? = nil,
        searchText: String? = nil
    ) async -> Result<CursorPage<PoolGamblerBet>, Error> {
        return await poolGamblerBetRepository.getPendingPoolGamblerBets(
            poolId: poolId, 
            gamblerId: gamblerId,
            next: next,
            searchText: searchText
        )
    }
}
